openpgp/key_generation.go

```diff
@@ -143,6 +142,16 @@ func NewEntity(name, comment, email string, config *packet.Config) (*Entity, err
 		e.Identities[uid.Id].SelfSignature.PreferredSymmetric = []uint8{uint8(config.DefaultCipher)}
 	}
 
+	// And for DefaultMode.
+	if config != nil && config.AEAD().DefaultMode != 0 && config.IsAEADEnabled() {
```

Comment:
Right, it doesn't panic, but we still need to check if `config` is nil and only after we can access `config.AEAD()` (or either change `.AEAD()` to return a pointer). But still, why is the current state asymmetric with the above (other than the `config.IsAEADEnabled` check)?

Reply:
So - if we change it to a pointer, and return `nil` by default, the symmetry is more clear if you write it like:

```go
var config *packet.Config
var aeadConfig *packet.AEADConfig

if config != nil && config.DefaultCipher != 0 {
	...
}

aeadConfig := config.AEAD()
if aeadConfig != nil && aeadConfig.DefaultMode != 0 {
	...
}
```
whereas the current code is instead:
```go
if config != nil && aeadConfig.DefaultMode != 0 {
	...
}
```
And if we take `aeadConfig != nil` to mean that AEAD is enabled, then even the `IsAEADEnabled` check isn't necessary anymore.

Reply:
Hmm, the code looks symmetric but the behavior doesn't, since `aeadConfig` is an attribute of `config`. 
For instance, if `config == nil`, then `...SelfSignature.PreferredAEAD` would still be set (assuming `nil config` has AEAD enabled by default, as per you comment in `config.go`). Do we want that? Assuming yes, then shouldn't we be doing the same with the other 2 bad guys? (`PreferredSymmetric` and `PreferredHash`). 
On a side note, this function is way too long (>160 lines).